Userland/Libraries/LibWeb/HTML/Parser/HTMLToken.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <andrew@ladybird.org>>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+public class HTMLToken {
+    public struct Position {
+        var line = UInt()
+        var column = UInt()
+        var byteOffset = UInt()
+    }
+
+    public struct Attribute {
+        var prefix: String?
+        var localName: String
+        var namespace_: String?
+        var value: String
+        var nameStartPosition: Position
+        var nameEndPosition: Position
+        var valueStartPosition: Position
+        var valueEndPosition: Position
+    }
+
+    public enum TokenType {
+        case Invalid
```

Comment:
Are we doing upper camel case for enum variants? Foundation and the docs seem to use lower camel case.

Reply:
wait, really? ew. why would I want my variants to start with a lower case ;-;. sigh. I'll follow the 'official' style though.

Reply:
Err...why?
we ignored plenty of C++ style rules, if it looks better, it looks better.

Reply:
IMO, if these are going to be visible from C++ land, keep the capitalisation consistent with that.

Reply:
eh, all right. I'll leave this as is until we get more swift files in the repo.